File: Bread.cpp
//
// Created by NandM2 on 12/10/2021.
//

#include "Bread.h"

#include <utility>

Bread::Bread(std::string variety) : BakedGood(4.50), variety(std::move(variety)) {}

std::string Bread::ToString() const {
    return variety + " bread " + BakedGood::ToString();
}

double Bread::DiscountedPrice(int quantity) {
    return (quantity - quantity/3) * BakedGood::getBasePrice();
}

File: Dart/bakery_checkout/bin/bread.dart
import 'baked_good.dart';

class Bread extends BakedGood {
  final String _variety;

  Bread(this._variety) : super(4.50);

  @override
  String toString() {
    return "$_variety bread " + super.toString();
  }

  @override
  double discountedPrice(int quantity) {
    return (quantity - quantity ~/ 3) * super.basePrice;
  }
}
